src/nni_manager/training_service/kubernetes/kubeflow/kubeflowTrainingService.ts

```diff
@@ -215,18 +215,28 @@ class KubeflowTrainingService extends KubernetesTrainingService implements Kuber
             if (this.azureStorageClient === undefined) {
                 throw new Error('azureStorageClient is not initialized');
             }
+            let retryCount: number = 1;
             try {
-                //upload local files, including scripts for running the trial and configuration (e.g., hyperparameters) for the trial, to azure storage
-                await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
-                                                                `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
-                                                                `${trialLocalTempFolder}`);
-                //upload code files to azure storage
-                await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
-                                                                `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
-                                                                `${this.kubeflowTrialConfig.codeDir}`);
-
-                trialJobOutputUrl = `https://${this.azureStorageAccountName}.file.core.windows.net/${this.azureStorageShare}` + 
-                                    `/${path.join('nni', getExperimentId(), trialJobId, 'output')}`;
+                while (retryCount >= 0) {
+                    //upload local files, including scripts for running the trial and configuration (e.g., hyperparameters) for the trial, to azure storage
+                    let resultUploadDir1: boolean = await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
+                        `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
+                        `${trialLocalTempFolder}`);
+                    //upload code files to azure storage
+                    let resultUploadDir2: boolean = await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
+                        `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
+                        `${this.kubeflowTrialConfig.codeDir}`);
+                    if (resultUploadDir1 && resultUploadDir2) {
+                        trialJobOutputUrl = `https://${this.azureStorageAccountName}.file.core.windows.net/${this.azureStorageShare}` + 
+                        `/${path.join('nni', getExperimentId(), trialJobId, 'output')}`;
+                        break;
+                    } else {
+                        //wait for 5 seconds to re-upload files
+                        await delay(5000);
+                        this.log.info('Re-upload files to azure-storage');
+                        retryCount -= 1;
+                    }
+                } 
```

Comment:
these code is duplicated with frameworklauncher training service, consider to do a refactor to remove dup code.

Reply:
fixed, extract duplicated code to kubernetesTrainingService.